tests need python modules that are not yet installed, so point the built ones.
Reported upstream: https://bugreports.qt.io/browse/PYSIDE-1429 (shiboken2)
Updated for shiboken6 : use a shiboken6's temporary python module.

--- a/sources/shiboken6/tests/CMakeLists.txt.ORIG	2021-05-05 08:27:28.000000000 +0200
+++ b/sources/shiboken6/tests/CMakeLists.txt	2021-05-14 17:37:25.572549382 +0200
@@ -52,7 +52,7 @@
     set(test_name "${CMAKE_MATCH_1}_${CMAKE_MATCH_3}")
     list(FIND test_blacklist ${test_name} expect_fail)
     add_test(${test_name} ${PYTHON_EXECUTABLE} ${test_file})
-    set_tests_properties(${test_name} PROPERTIES ENVIRONMENT "BUILD_DIR=${BUILD_DIR}")
+    set_tests_properties(${test_name} PROPERTIES ENVIRONMENT "BUILD_DIR=${BUILD_DIR};PYTHONPATH=${minimal_BINARY_DIR}/../../tmpmodule:${minimal_BINARY_DIR}:${sample_BINARY_DIR}:${other_BINARY_DIR}:${smart_BINARY_DIR}")
     set_tests_properties(${test_name} PROPERTIES TIMEOUT ${CTEST_TESTING_TIMEOUT})
     if (${expect_fail} GREATER -1)
         set_tests_properties(${test_name} PROPERTIES WILL_FAIL TRUE)
